IM_calculation/Advanced_IM/scripts/overview_adv_IM_status.py

```diff
@@ -0,0 +1,135 @@
+    for event_status_file in event_status_files:
+        if type == "sim":
+            event_name = Path(event_status_file).parents[2].name
+        else:
+            event_name = Path(event_status_file).parent.name
+        event_status_df = pd.read_csv(event_status_file)
+        event_status_dict[event_name] = {"total": 0}
+        for status in ALL_STATUS:
+
+            count = event_status_df["status"].value_counts().get(status, 0)
+            event_status_dict[event_name][status] = count
+            event_status_dict[event_name]["total"] += count
+
+    overall_status_df = pd.DataFrame.from_dict(
+        event_status_dict, orient="index", columns=["total"] + ALL_STATUS
+    )
+
+    overall_status_df.to_csv(outdir / f"overall_status_{model}_{type}.csv")
+
+    not_finished_df = overall_status_df.loc[
+        overall_status_df["total"] != overall_status_df["finished"]
+    ]
+    print("------------")
+    print(
+        overall_status_df.sum(axis=0).to_string(header=None)
+    )  # .to_string(header=None) removes dtype: int64
+    print("------------")
+    if len(not_finished_df) > 0:
+        print(
+            f"{len(not_finished_df)} incomplete/unsuccessful event(s) found (Total: {len(overall_status_df)})"
+        )
+        if verbose:
+            print(not_finished_df)
+    else:
+        print(f"All completed successfully! (Total: {len(overall_status_df)})")
+    return overall_status_df
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "sim_root",
+        type=Path,
+        help="Root path of GM sim.ie. Parent directory of Runs or ObservedGroundMotions",
+    )
+    parser.add_argument("model", help="Name of the advanced IM model")
+    parser.add_argument(
+        "type",
+        choices=TYPES,
+        default="custom",
+        help="Type of data. If neither sim or obs, choose custom and provide --pattern",
+    )
+    parser.add_argument(
+        "--outdir", type=Path, default=Path.cwd(), help="Output directory"
+    )
+    parser.add_argument(
+        "--pattern",
+        default=None,
+        help='Status file search pattern for custom type. eg.) "Runs/*/*/IM_calc"',
+    )
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        default=False,
+        help="Print unfinished events summary",
+    )
+
+    args = parser.parse_args()
+
+    if not args.sim_root.is_dir():
+        raise AssertionError("Provide a valid directory path")
+
+    if args.type == "custom":
+        if args.pattern is None:
+            raise AssertionError("Type custom must have a search pattern")
+        else:
+            args.pattern = args.pattern.strip(
+                "/"
+            )  # remove leading or trailing "/" (if any)
+    else:
+        if args.pattern is not None:
+            print(
+                f"Warning: Type {args.type} uses pre-defined search pattern - supplied one ignored"
+            )
+
+    process(
```

Comment:
Would aggregate_status_files be a better function name?

Reply:
Sure - to be changed in the next commit

Reply:
Done